src/main/java/com/aws/greengrass/deployment/ConnectivityValidator.java

```diff
@@ -0,0 +1,230 @@
+    private final SecurityService securityService;
+    @Setter(AccessLevel.PACKAGE)  // for unit tests
+    private DeviceConfiguration deploymentConfiguration;
+
+    /**
+     * Constructor.
+     *
+     * @param kernel           kernel to get current config and context from
+     * @param context          temporary context used for deployment configuration
+     * @param mqttClient       Used to create AWS Mqtt Connection Client
+     * @param factory          Used by integration tests to set mock
+     * @param deploymentConfig Map of the configs to be merged in and validated against
+     * @param timestamp        time stamp of the deployment
+     */
+    public ConnectivityValidator(Kernel kernel, Context context, MqttClient mqttClient,
+                                 GreengrassServiceClientFactory factory, Map<String, Object> deploymentConfig,
+                                 long timestamp) {
+        this.mqttClient = mqttClient;
+        this.factory = factory;
+        securityService = kernel.getContext().get(SecurityService.class);
+
+        Configuration config = new Configuration(context);
+        // Copy the current device configuration Map to preserve time stamps
+        config.copyFrom(kernel.getConfig());
+        // Attempt to merge the deployment configs using the deployment time stamp
+        config.mergeMap(timestamp, deploymentConfig);
+        try {
+            config.waitConfigUpdateComplete();
+        } catch (InterruptedException e) {
+            logger.atError().log("Wait for deployment config update complete was interrupted");
+            Thread.currentThread().interrupt();
+        }
+        deploymentConfiguration = new DeviceConfiguration(config, kernel.getKernelCommandLine());
+        /*
+         * Need to set security service due to plugin dependency workaround
+         * after removing Kernel from DeviceConfiguration
+         */
+        deploymentConfiguration.setSecurityService(securityService);
+    }
+
+    /**
+     * Check if there's any meaningful changes that would require MQTT validation.
+     *
+     * @param currentDeviceConfiguration current configs the device is using
+     */
+    public boolean mqttClientNeedsValidation(DeviceConfiguration currentDeviceConfiguration) {
+        if (!deploymentConfiguration.isDeviceConfiguredToTalkToCloud()) {
+            return false;
+        }
+        return networkPoxyHasChanged(currentDeviceConfiguration) || awsRegionHasChanged(currentDeviceConfiguration)
+                || mqttHasChanged(currentDeviceConfiguration) || iotDataEndpointHasChanged(currentDeviceConfiguration);
+    }
+
+    /**
+     * Check if there's any meaningful changes that would require HTTP validation.
+     *
+     * @param currentDeviceConfiguration current configs the device is using
+     */
+    public boolean httpClientNeedsValidation(DeviceConfiguration currentDeviceConfiguration) {
+        if (!deploymentConfiguration.isDeviceConfiguredToTalkToCloud()) {
```

Comment:
no reason to check for configured to talk to cloud 2 times. Just check it once before asking if the mqtt and http clients need validation.